vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -117,4 +117,6 @@ def build(self):
             ],
             color="grey",
             parent_className="loading-container",
+            delay_show=500,
```

Comment:
@AnnMarieW—I have a quick question about the `delay_show`. I've read through your documentation (great examples, by the way! 🚀 ) and your original PR on Dash. From what I understand, it reduces the flickering because the loading spinner is only shown after 500ms. If the loading time is short, the loading spinner might not even appear, did I understand that correctly? So, I guess we need to find the sweet spot in terms of time here.

When running the demo dashboards (`hatch run example demo`) with `delay_show=500`, it seems to flicker a bit more now as the loading spinner is displayed for a shorter time. Are there any other advantages for the `delay` arguments that might not be visually clear from the Vizro examples?

Reply:
Yes, I agree! I've tried both with our example demo 😄 but faced the issue you described. The `delay_hide` artificially delays, which we don't want from a UX perspective. The faster, the better. 

The `delay_show` sounds lovely, but I couldn't find the right time, so it looks like it's flickering more because the loader appears for a shorter amount of time. I think J. prefers to have the loaders visible, even briefly, to let people know something is happening.

Reply:
Yes, finding the sweet spot can be a little tricky.

Note that you can include both the `delay_show` and the `delay_hide`.  

I'll make the change to also include `delay_hide=500`  This makes it so that once the spinner is triggered it will show for a min of 500ms.  So that should help with the flash when the callback takes around 600ms

Reply:
As aligned with the rest, let's remove `delay_show` and `delay_hide` and define the overlay_style only 👍